### PR TITLE
countdown module created

### PR DESCRIPTION
**Pull request by 418m**

### CODE REVIEW

File: countdown/module.py [hunk 1]
```diff
@@ -0,0 +1,256 @@
...
+        except dateutil.parser.ParserError:
+            await ctx.reply(
+                _(
+                    ctx,
+                    f"I don't know how to parse `{datetime_str}`, pleasy try again.",
+                ).format(datetime_str=datetime_str)
+            )
+            return
+
+        if date < datetime.now():
```
Comment (ArcasCZ) [inline countdown/module.py:144]: Since you are throwing ValueError, you might also use try-except block.

File: countdown/module.py [hunk 2]
```diff
@@ -0,0 +1,256 @@
...
+guild_log = logger.Guild.logger()
+
+
+class Countdown(commands.Cog):
+    """Countdown"""
+
+    def __init__(self, bot):
+        self.bot = bot
+
+    async def _process_text(self, ctx: commands.Context, text: Optional[str]):
```
Comment (ArcasCZ) [inline countdown/module.py:24]: This should be a utils function in pumpkin-py core. I think I've seen this kind of code in other modules / pumpkin-py repos.

**Review by ArcasCZ — COMMENTED** [review]
Overall the code looks good, it's easy to read and I don't see any spaghetti code there.

File: countdown/database.py [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+        guild_id: ID of the guild.
+        author_id: User ID of reminder author.
+        permalink: Message URL.
+        message: Reminder text (None if empty).
+        origin_date: Date of creation.
+        countdown_date: Date for reminding
+        status: Status of reminder
+    """
+
+    __tablename__ = "fun_countdown_countdownitem"
```
Comment (sinus-x) [inline countdown/database.py:26]: This could be just `fun_countdown_item`

File: countdown/database.py [hunk 4]
```diff
@@ -0,0 +1,176 @@
...
+            session.query(CountdownItem)
+            .filter_by(author_id=author_id, name=name)
+            .one_or_none()
+        )
+        return query
+
+    @staticmethod
+    def get_all(
+        guild: discord.Guild = None,
+        idx: int = None,
```
Comment (sinus-x) [inline countdown/database.py:93]: 1/ Do we need to filer by idx?
2/ I'd prefer if database files worked directly with `guild_id`, not DIscord objects. This is different from how the rest of the codebase is done.

File: countdown/database.py [hunk 5]
```diff
@@ -0,0 +1,176 @@
...
+        author: discord.Member = None,
```
Comment (sinus-x) [inline countdown/database.py:94]: Same here, IDs instead of objects.

File: countdown/module.py (hunk 2 above)
Comment (sinus-x) [inline countdown/module.py:24]: I'd like to see some docstring, describing what the function does to the text.

File: countdown/module.py [hunk 6]
```diff
@@ -0,0 +1,256 @@
...
+        if text is not None and len(text) > 1024:
+            text = text[:1024]
+            text = text[:-3] + "```" if text.count("```") % 2 != 0 else text
+
+        return text
+
+    async def _get_member(self, user_id: int, guild_id: int = 0):
```
[the remark this answers is not in the document]
Comment (sinus-x) [reply countdown/module.py:31]: Yep, it can go.

File: countdown/module.py [hunk 7]
```diff
@@ -0,0 +1,256 @@
...
+        embed.add_field(name=_(ctx, "URL"), value=countdown.permalink)
+
+        return embed
+
+    async def _send_countdown_list(self, ctx, query):
+        countdowns = []
+
+        for item in query:
+
+            countdown = CountdownDummy()
```
Comment (sinus-x) [inline countdown/module.py:81]: You don't need to create new class, in Python you can just add attributes into new `countdown = object()`

**Review by sinus-x — CHANGES_REQUESTED** [review]
The code looks good, I have some minor requests.

File: countdown/database.py [hunk 8]
```diff
@@ -0,0 +1,171 @@
...
+            session.query(CountdownItem)
+            .filter_by(author_id=author_id, name=name)
+            .one_or_none()
+        )
+        return query
+
+    @staticmethod
+    def get_all(
+        guild_id: discord.Guild = None,
+        author_id: discord.Member = None,
```
Comment (sinus-x) [inline countdown/database.py:94]: The type hint should be `: int` now.

**Review by sinus-x — CHANGES_REQUESTED** [review]

File: countdown/database.py (hunk 8 above)
Comment (418m) [reply countdown/database.py]: Fixed

**Review by sinus-x — APPROVED** [review]
Checks have passed, I'm gonna assume it is fine.

**Pull request merged**